### PR TITLE
Feature/credential service basic constructor

### PR DESCRIPTION
**Pull request by james-toussaint**

### CODE REVIEW

File: src/test/java/com/iexec/common/chain/CredentialsAbstractServiceTests.java [hunk 1]
```diff
@@ -46,11 +47,43 @@ public CredentialsServiceStub(String walletPassword, String walletPath) throws E
...
+        public CredentialsServiceStub(Credentials credentials) throws Exception {
+            super(credentials);
+        }
+
         public CredentialsServiceStub() throws Exception {
             super();
         }
     }
 
+    // begin constructor (Credentials)
```
Comment (mcornaton) [inline src/test/java/com/iexec/common/chain/CredentialsAbstractServiceTests.java]: What about `// region` comments to enable smart folding?

File: src/main/java/com/iexec/common/chain/CredentialsAbstractService.java [hunk 2]
```diff
@@ -50,6 +45,17 @@ public CredentialsAbstractService() throws Exception {
         }
     }
 
+    public CredentialsAbstractService(Credentials credentials) {
```
Comment (jbern0rd) [inline src/main/java/com/iexec/common/chain/CredentialsAbstractService.java]: Constructor in abstract classes should never be `public` as the constructor will never be called directly.
See Sonar alert and new code smell.

https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=352&id=com.iexec.common%3Aiexec-common

**Review by jbern0rd — APPROVED** [review]

File: src/test/java/com/iexec/common/chain/CredentialsAbstractServiceTests.java (hunk 1 above)
Comment (james-toussaint) [reply src/test/java/com/iexec/common/chain/CredentialsAbstractServiceTests.java]: Updated here https://github.com/iExecBlockchainComputing/iexec-common/pull/352/commits/d62f3848167c6cfe05a57aac131253dbad855d19, thanks :)

File: src/main/java/com/iexec/common/chain/CredentialsAbstractService.java (hunk 2 above)
Comment (james-toussaint) [reply src/main/java/com/iexec/common/chain/CredentialsAbstractService.java]: Updated here https://github.com/iExecBlockchainComputing/iexec-common/pull/352/commits/e57d5c61791cb1523c2b6bd010fe056ee5cc52d6, thanks for the spot!

**Pull request merged**